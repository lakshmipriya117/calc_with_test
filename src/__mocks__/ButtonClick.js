class ButtonClick{
  static clickEvent(value){
    if(value == "C"){
      return 0;
    }
    return Number(value);
  }
}

export default ButtonClick
