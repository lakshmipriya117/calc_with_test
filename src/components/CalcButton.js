/**
 * The CalcButton class is component that renders as a button that holds
 * the appropriate int of operation values
 *
 */
import React,{Component} from 'react'

export default class CalcButton extends Component{
  /**
  * This method is used to render and display the button component
  *
  * @return input - button element displaying the appropriate value
  *
  */
  render(){
    return(
      <input className="calc-btn" type="button" value={this.props.label} onClick={this.clickHandler}/>
    )
  }
  /**
  * This method is used to render and display the button component
  *
  * @param e - event payload
  *
  */
  clickHandler = (e) => {
    this.props.clickedHandler(e.target.value);
  }
}
