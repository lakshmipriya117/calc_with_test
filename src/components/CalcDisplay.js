/**
 * The CalcDisplay class is a component that renders as a textarea that holds
 * the appropriate calculated value based on the buttons clicked
 *
 */

import React,{Component} from 'react'

export default class CalcDisplay extends Component{
  constructor(props){
    super(props);
    this.props = props;

  }
  /**
  * This method is used to render and display the textarea component
  *
  * @return textarea -  element displaying the appropriate calculated value
  *
  */
  render(){
    return(
      <textarea className="calc-display" value={this.props.value}/>
    )
  }
}
