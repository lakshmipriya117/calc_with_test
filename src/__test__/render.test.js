import renderer from 'react-test-renderer';
import CalcDisplay from '../components/CalcDisplay'
import CalcButton from '../components/CalcButton'
import ButtonClick  from '../__mocks__/ButtonClick'
import React from 'react';

/*
* test to check whether the DOM and display rendered correct based on the buttons clicked
*
*/
describe('check the rendering of components', () => {
  it('CalcDisplay component renders correctly', () => {
    const tree = renderer
      .create(<CalcDisplay value="27"></CalcDisplay>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CalcDisplay component renders correctly', () => {
    const tree = renderer
      .create(<CalcDisplay value="{ButtonClick.clickEvent('2')}"></CalcDisplay>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CalcDisplay component renders correctly', () => {
    const tree = renderer
      .create(<CalcDisplay value="{ButtonClick.clickEvent('C')}"></CalcDisplay>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CalcButton component renders correctly', () => {
    const tree = renderer
      .create(<CalcButton label={6} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
