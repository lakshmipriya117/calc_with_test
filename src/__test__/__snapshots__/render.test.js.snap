// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the rendering of components CalcButton component renders correctly 1`] = `
<input
  className="calc-btn"
  onClick={[Function]}
  type="button"
  value={6}
/>
`;

exports[`check the rendering of components CalcDisplay component renders correctly 1`] = `
<textarea
  className="calc-display"
  value="27"
/>
`;

exports[`check the rendering of components CalcDisplay component renders correctly 2`] = `
<textarea
  className="calc-display"
  value="{ButtonClick.clickEvent('2')}"
/>
`;

exports[`check the rendering of components CalcDisplay component renders correctly 3`] = `
<textarea
  className="calc-display"
  value="{ButtonClick.clickEvent('C')}"
/>
`;
